src/main/java/org/broadinstitute/hellbender/cmdline/GATKPlugin/GATKCommandLinePluginDescriptor.java

```diff
@@ -0,0 +1,142 @@
+     *                       by getPluginClass
+     * @return the instantiated object that will be used by the command line parser
+     * as an argument source
+     * @throws IllegalAccessException
+     * @throws InstantiationException
+     */
+    public abstract Object addInstance(Class<?> pluggableClass)
+            throws IllegalAccessException, InstantiationException;
+
+    /**
+     * Return the allowable values for the String argument specified by
+     * longArgName. Called by the command line parser to generate a usage
+     * string. If the value is unrecognized, the implementation should
+     * throw IllegalArgumentException.
+     *
+     * @param longArgName
+     * @return Set<String> of allowable values</String>, or null if any value is allowed
+     */
+    public abstract Set<String> getAllowedStringValues(String longArgName);
+
+    /**
+     * Called by the command line parser when an argument value from the class
+     * targetPluginClass has been seen on the command line. Returns true if the
+     * argument is allowed (ie., an instance of the dependent parent plugin class
+     * was actually seen), otherwise false.
+     *
+     * @param targetPluginClass
+     * @return true if the plugin for this class was specified on the command line
+     */
+    public abstract boolean isDependentArgumentAllowed(Class<?> targetPluginClass);
+
+    /**
+     * It is the descriptor's job to contain an argument list which will be populated
+     * by the command line parser with the name of each plugin specified on the command line.
+     *
+     * This method is called after all command line arguments have been processed to allow
+     * the descriptor to validate the plugin arguments specified, and to reduce the list of
+     * plugin instances to those actually seen on the command line.
+     *
+     * Implementations of this method should validate that all of the values that have been
+     * specified on the command line have a corresponding plugin instance (this will detect
+     * a user-specified value for which there is no corresponding plugin class).
+     * @throws UserException.CommandLineException if a plugin value has been specified that
+     * has no corresponding plugin instance (i.e., the plugin class corresponding to the name
+     * was not discovered)
+     */
+    public abstract void validateArguments() throws UserException.CommandLineException;
+
+    /**
+     * Given a consumer of a list of plugin objects of type T, call the consumer's
+     * accept method with the list actual instances accumulated by the plugin (those
+     * actually specified on the command line). This gives clients of the plugin
+     * objects a chance to retrieve the actual list.
+     *
+     * @param consumer function that accepts a collection of plugin instances
+     */
+    public abstract void getInstances(Consumer<Collection<T>> consumer);
```

Comment:
Should this getter just return `Collection<T>`, instead of taking a `Consumer` and returning nothing?